### PR TITLE
AIM: Add key to step outside containers; mark other pane's container; less obtrusive popups

### PR DESCRIPTION
**Pull request by Kamayana**
#### Summary
Interface "AIM: Add key to step outside containers; AIM: mark container that the other pane is looking inside"

#### Purpose of change
#65560 added multi-container support to the advanced inventory manager, but there were a few minor issues with the implementation:
1. The header minimap's [C] color changing behavior would act erratically between the panes if they were both looking in containers.
2. Doubling the container location with AIM_CONTAINER2 is functional, but it has a pretty large code footprint for functionality that could be implemented more concisely.

While writing a followup to those, I also became increasingly aware of some issues with how the AIM has handled containers since I reworked it months ago:

3. There's no way to step outside of a container to its parent, which is annoying if you go too far or just want to step in for a moment.
4. The way items are hidden between panels makes it difficult to work with nested containers now that that's possible.
5. The popups whenever an item can't be moved can be obnoxious, especially since only specific keys can close them.
6. You can't move all if both panes are in containers.
7. Containers can show the wrong weight/volume in the header.

#### Describe the solution
1 and 2: Replaced the AIM_CONTAINER2 implementation with a single check in `action_move_item()`. This both fixed the minimap and reduced the footprint.

3: Added a new key and location to the AIM: `<X>`, for "eXit to parent". This key will step outside of a container to its parent container or location. Containers are now marked with inward facing carets `>C<`, with exiting marked with outward facing ones `<X>`. The minimap highlighting rules for `<X>` and `>C<` show when either action is available.

4: Instead of hiding a container from the other pane, it's now highlighted in a special way to clearly indicate that the other pane is inside.

5: Almost all popups were changed to use `popup_getkey()`, which is simply a popup that can be closed with any keypress. I also removed the exclamation points from most of them because they rarely warranted the excitement ("This is too heavy!" is good, I kept that one).

6 and 7: Fixed.

#### Describe alternatives you've considered
- Replace most popups with simple messages to the sidebar, instead. Tried it, but it felt like it de-emphasized them too much.
- Tried many, many ways of highlighting the container. Went with the one I felt had the best clarity and aesthetics.
- There are some additional things I'd like to do down the line, but I'm trying to tone down my mega PRs.

#### Testing
Lots of various testing throughout, swapping, moving to and from containers, etc.. stuffed many plastic bags within one another and spammed the AIM to trigger `print_items` a ton: zero CPU impact.

#### Additional context
image

### CODE REVIEW

File: src/advanced_inv.cpp [hunk 1]
```diff
@@ -958,31 +958,20 @@ bool advanced_inventory::move_all_items()
...
-    // Check first if the destination area still has enough room for moving all.
-    const units::volume &src_volume = spane.in_vehicle() ? sarea.volume_veh : sarea.volume;
-    units::volume dest_volume_free;
-    if( dpane.container ) {
-        dest_volume_free = dpane.container->get_remaining_capacity();
-    } else {
-        dest_volume_free = darea.free_volume( dpane.in_vehicle() );
-    }
-    if( !is_processing() && src_volume > dest_volume_free &&
-        !query_yn( _( "There isn't enough room.  Attempt to move as much as you can?" ) ) ) {
```
Comment (Qrox) [inline src/advanced_inv.cpp:983]: Why is this removed?

**Comment by ashGlaw** [comment]
Those issues have been bugging me for a while too. Thanks for improving our AIM

File: src/activity_actor.cpp [hunk 2]
```diff
@@ -4184,10 +4184,9 @@ void insert_item_activity_actor::finish( player_activity &act, Character &who )
             int charges = std::min( holstered_item.second, holster.max_charges_by_parent_recursive( it ) );
 
             if( charges > 0 && holster->can_contain_partial_directly( it ) ) {
-                int result = holster->fill_with( it, charges,
-                                                 /*unseal_pockets=*/true,
-                                                 /*allow_sealed=*/true,
-                                                 /*ignore_settings*/true );
+                const bool into_bottom = holstered_item.first.parent_item() &&
+                                         holster == holstered_item.first.parent_item().parent_item();
+                int result = holster->fill_with( it, charges, true, true, true, into_bottom );
```
[the remark this answers is not in the document]
Comment (Kamayana) [reply src/activity_actor.cpp:4189]: Ah, good to know the reasoning behind those comments, I'll re-add them.

File: src/advanced_inv.cpp (hunk 1 above)
Comment (Kamayana) [reply src/advanced_inv.cpp:983]: Accidental leftover from a non-pushed change, good catch.

File: src/character_inventory.cpp [hunk 3]
```diff
@@ -75,7 +75,12 @@ void Character::handle_contents_changed( const std::vector<item_location> &conta
             if( pocket && handle_drop && pocket->will_spill() ) {
                 // the pocket's contents (with a larger depth value) are not
                 // inside `sorted_containers` and can be safely disposed of.
-                pocket->handle_liquid_or_spill( *this, /*avoid=*/&*loc );
+                if( loc.where_recursive() == item_location::type::character ) {
+                    pocket->handle_liquid_or_spill( *this, /*avoid=*/&*loc );
+                } else {
+                    add_msg_if_player( m_info, _( "The %s spills its contents!" ), loc->tname() );
+                    pocket->spill_contents( loc.pos_bub().raw() );
```
[the remark this answers is not in the document]
Comment (Kamayana) [reply src/character_inventory.cpp:82]: The spill changes were originally to fix a crash, but on removing the changes I'm unable to replicate the crash, so I guess they're unnecessary and I can revert them.

File: src/activity_actor.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (andrei8l) [reply src/activity_actor.cpp]: IDEs can expand those so they're not really needed

this is from clangd in vscode, after removing the comments
Screenshot from 2023-07-20 19-17-01

[the remark this answers is not in the document]
Comment (Qrox) [reply src/activity_actor.cpp]: Huh, didn't know IDEs could do that nowadays. But I still think it's important to have those information in the code itself in case someone is using a less smart IDE (or no IDE at all like me).

File: data/raw/keybindings.json [hunk 4]
```diff
@@ -3721,6 +3721,17 @@
       { "input_method": "keyboard_code", "key": "c", "mod": [ "shift" ] }
     ]
   },
+  {
+    "type": "keybinding",
```
Comment (Maleclypse) [inline data/raw/keybindings.json:3725]:
```suggestion
    "type": "keybinding"
```
Resetting test suite

File: data/raw/keybindings.json [hunk 5]
```diff
@@ -3721,6 +3721,17 @@
       { "input_method": "keyboard_code", "key": "c", "mod": [ "shift" ] }
     ]
   },
+  {
+    "type": "keybinding"
```
Comment (Maleclypse) [inline data/raw/keybindings.json:3725]:
```suggestion
    "type": "keybinding",
```
